File: bom-analyzer/bomlib/env.py
"""Minimal .env reader so the project stays dependency-free."""

import os


def load_env(path=None):
    """Read KEY=VALUE lines into os.environ.

    Values already present in the environment win, which lets shell exports and
    hosting-platform config override the file.
    """
    target = path or os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), '.env')
    parsed = {}
    try:
        with open(target, 'r', encoding='utf-8') as handle:
            raw = handle.read()
    except FileNotFoundError:
        return parsed

    for line in raw.splitlines():
        trimmed = line.strip()
        if not trimmed or trimmed.startswith('#'):
            continue
        if '=' not in trimmed:
            continue
        key, _, value = trimmed.partition('=')
        key = key.strip()
        if not key:
            continue
        value = value.strip()
        if len(value) > 1 and value[0] == value[-1] and value[0] in ('"', "'"):
            value = value[1:-1]
        parsed[key] = value
        os.environ.setdefault(key, value)
    return parsed

File: bom-analyzer/lib/env.js
'use strict';

const fs = require('fs');
const path = require('path');

// Minimal .env reader so the project stays dependency-free. Values already
// present in process.env win, which lets shell exports and hosting-platform
// config override the file.
function loadEnv(file) {
  const target = file || path.join(__dirname, '..', '.env');
  let raw;
  try {
    raw = fs.readFileSync(target, 'utf8');
  } catch (err) {
    if (err.code === 'ENOENT') return {};
    throw err;
  }

  const parsed = {};
  for (const line of raw.split(/\r?\n/)) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith('#')) continue;
    const eq = trimmed.indexOf('=');
    if (eq === -1) continue;
    const key = trimmed.slice(0, eq).trim();
    if (!key) continue;
    let value = trimmed.slice(eq + 1).trim();
    if (
      (value.startsWith('"') && value.endsWith('"') && value.length > 1) ||
      (value.startsWith("'") && value.endsWith("'") && value.length > 1)
    ) {
      value = value.slice(1, -1);
    }
    parsed[key] = value;
    if (process.env[key] === undefined) process.env[key] = value;
  }
  return parsed;
}

module.exports = { loadEnv };
